pkg/api/userconfig/compute.go

```diff
@@ -212,6 +221,13 @@ var apiComputeFieldValidation = &cr.StructFieldValidation{
 					Min: k8sresource.MustParse("0"),
 				}),
 			},
+			&cr.StructFieldValidation{
+				StructField: "GPU",
+				Int64PtrValidation: &cr.Int64PtrValidation{
+					Default:     nil,
+					GreaterThan: util.Int64Ptr(0),
+				},
+			},
 		},
 	},
 }
```

Comment:
The GPU field should be added to `(apiCompute *APICompute) ID()` and `(apiCompute *APICompute) IDWithoutReplicas()`